spec/rubocop/cop/style/single_argument_dig_spec.rb

```diff
@@ -0,0 +1,151 @@
+          RUBY
+        end
+
+        it 'does not register an offense with variables as arguments' do
+          expect_no_offenses(<<~RUBY)
+            var1 = :key1
+            var2 = :key2
+            { key1: { key2: 'value' } }.dig(var1, var2)
+          RUBY
+        end
+
+        it 'does not register an offense with a variable as the dig caller' do
+          expect_no_offenses(<<~RUBY)
+            hash = { key1: { key2: 'value' } }
+            hash.dig(:key1, :key2)
+          RUBY
+        end
+      end
+
+      it 'does not register an offense when using dig with splat operator' do
+        expect_no_offenses(<<~RUBY)
+          { key1: { key2: 'value' } }.dig(*%i[key1 key2])
+        RUBY
+      end
+    end
+
+    describe 'array' do
+      describe 'with single argument' do
+        it 'registers an offense with literal argument' do
+          expect_offense(<<~RUBY)
+            [1, 2].dig(0)
+            ^^^^^^^^^^^^^ Use `[1, 2][0]` instead of `[1, 2].dig(0)`.
+          RUBY
+        end
+
+        it 'registers an offense with a variable as an argument' do
+          expect_offense(<<~RUBY)
+            index = 0
+            [1, 2].dig(index)
+            ^^^^^^^^^^^^^^^^^ Use `[1, 2][index]` instead of `[1, 2].dig(index)`.
+          RUBY
+        end
+
+        it 'registers an offense with a variable as the dig caller' do
+          expect_offense(<<~RUBY)
+            array = [1, 2]
+            array.dig(0)
+            ^^^^^^^^^^^^ Use `array[0]` instead of `array.dig(0)`.
+          RUBY
+        end
+
+        it 'autocorrect using dig to [] with a literal argument' do
+          expect(autocorrect_source('[1, 2].dig(0)'))
+            .to eq('[1, 2][0]')
+        end
+
+        it 'autocorrect using dig to [] with a variable as an argument' do
+          expect(autocorrect_source('[1, 2].dig(var)'))
+            .to eq('[1, 2][var]')
+        end
+
+        it 'autocorrect using dig to [] with a variable as the dig caller' do
+          expect(autocorrect_source('array.dig(var)'))
+            .to eq('array[var]')
+        end
+      end
+
+      describe 'with multiple arguments' do
+        it 'does not register an offense with literal arguments' do
+          expect_no_offenses(<<~RUBY)
+            [1, [2, [3]]].dig(1, 1)
+          RUBY
+        end
+
+        it 'does not register an offense with variables as arguments' do
+          expect_no_offenses(<<~RUBY)
+            var1 = 1
+            var2 = 1
+            [1, [2, [3]]].dig(var1, var2)
+          RUBY
+        end
+
+        it 'does not register an offense with a variable as the dig caller' do
+          expect_no_offenses(<<~RUBY)
+            array = [1, [2, [3]]]
+            array.dig(1, 1)
+          RUBY
+        end
+      end
+
+      it 'does not register an offense when using dig with splat operator' do
+        expect_no_offenses(<<~RUBY)
+          [1, [2, [3]]].dig(*[1, 1])
+        RUBY
+      end
+    end
+  end
+end
```

Comment:
What do you think about using more contexts in your tests?
https://rspec.rubystyle.guide/#use-contexts

Reply:
For example this:

```ruby
      it 'does not register an offense when using dig with splat operator' do
        expect_no_offenses(<<~RUBY)
          [1, [2, [3]]].dig(*[1, 1])
        RUBY
      end
```

should be something like this:

```ruby
    context 'when using dig with splat operator' do
      it 'does not register an offense' do
        expect_no_offenses(<<~RUBY)
          [1, [2, [3]]].dig(*[1, 1])
        RUBY
      end
    end
```